src/Microsoft.OData.Edm/Csdl/Serialization/EdmModelCsdlSchemaJsonWriter.cs

```diff
@@ -0,0 +1,1292 @@
+        {
+            // A schema is represented as a member of the document object whose name is the schema namespace.
+            this.jsonWriter.WritePropertyName(schema.Namespace);
+
+            //  Its value is an object
+            this.jsonWriter.WriteStartObject();
+
+            //  It MAY contain the members $Alias
+            this.jsonWriter.WriteOptionalProperty("$Alias", alias);
+        }
+
+        /// <summary>
+        /// Write $Annotations : {
+        /// </summary>
+        /// <param name="outOfLineAnnotations">The total out of line annotations.</param>
+        internal override void WriteOutOfLineAnnotationsBegin(IEnumerable<KeyValuePair<string, List<IEdmVocabularyAnnotation>>> outOfLineAnnotations)
+        {
+            // The value of $Annotations is an object with one member per annotation target.
+            this.jsonWriter.WritePropertyName("$Annotations");
+            this.jsonWriter.WriteStartObject();
+        }
+
+        /// <summary>
+        /// Write Annotations with External Targeting.
+        /// </summary>
+        /// <param name="annotationsForTarget">The annotation.</param>
+        internal override void WriteAnnotationsElementHeader(KeyValuePair<string, List<IEdmVocabularyAnnotation>> annotationsForTarget)
+        {
+            // The member name is a path identifying the annotation target, the member value is an object containing annotations for that target.
+            this.jsonWriter.WritePropertyName(annotationsForTarget.Key);
+            this.jsonWriter.WriteStartObject();
+        }
+
+        /// <summary>
+        /// Write $Annotations End: }
+        /// </summary>
+        /// <param name="outOfLineAnnotations">The total out of line annotations.</param>
+        internal override void WriteOutOfLineAnnotationsEnd(IEnumerable<KeyValuePair<string, List<IEdmVocabularyAnnotation>>> outOfLineAnnotations)
+        {
+            this.jsonWriter.WriteEndObject();
+        }
+
+        /// <summary>
+        /// Write structural property object.
+        /// </summary>
+        /// <param name="property">The Edm structural property.</param>
+        /// <param name="inlineType">Is line type or not.</param>
+        internal override void WriteStructuralPropertyElementHeader(IEdmStructuralProperty property, bool inlineType)
+        {
+            // Structural properties are represented as members of the object representing a structured type. The member name is the property name.
+            this.jsonWriter.WritePropertyName(property.Name);
+
+            // The member value is an object.
+            this.jsonWriter.WriteStartObject();
+
+            // The property object MAY contain the member $Kind with a string value of Property. This member SHOULD be omitted to reduce document size.
+            // So, we omit the $Kind for property.
+
+            // It MAY contain the member $Type & $Collection
+            if (inlineType)

```

Comment:
>inlineType [](start = 16, length = 10)

Do we omit writing the type if it is Edm.String?  Or is that done in WriteTypeReference?

Reply:
Yes.  Below is the syntax of the WriteTypeReference.
```C#
private void WriteTypeReference(IEdmTypeReference type, string defaultTypeName = "Edm.String")
```
If the input type is same as the "defaultTypeName", we will omit it.



---
In reply to: [443715884](https://github.com/OData/odata.net/pull/1775#discussion_r443715884) [](ancestors = 443715884)